src/axl_io.c

```diff
@@ -593,6 +640,12 @@ int axl_file_copy(const char* src_file, const char* dst_file, unsigned long buf_
             copying = 0;
             rc = AXL_FAILURE;
         }
+        total_copied += nread;
+
+        /* Possibly pause our transfer for unit tests */
+        while (total_copied >= pause_after) {
+            pause_after = axl_debug_pause_after();
+        }
```

Comment:
This code is a bit convoluted when thinking about what it does.  Also, it's a bit strange to call this function repeatedly even when the option is not enabled.  Was the goal of writing it this way to make it easier to break the look in a debugger or for some other reason?

Reply:
For example, I think the intent would be easier for someone to parse with something like:
```
if (pause_after != ULONG_MAX && total_copied >= pause_after) {
  while(1);
}
```

Reply:
It's written the way it is so you can pause and unpause a transfer (by setting a different offset).  The cost is a `getenv()` on every megabyte read, which shouldn't be a big deal.  As you said though, it would be better to just check for the existence of `AXL_DEBUG_PAUSE_AFTER` at the beginning of the transfer, and if it's set, then continue to poll getenv() like it's doing now, since during test we're always going to be setting that envvar before starting the transfer.

Reply:
I don't think there is an easy way for someone to change the value that would be returned from ```getenv("AXL_DEBUG_PAUSE_AFTER")``` though, right?  How were you thinking someone would change what ```axl_debug_pause_after()``` is returning?

Reply:
`export AXL_DEBUG_PAUSE_AFTER=<new value>`

Reply:
@adammoody you were right about this - if you export a envar after a program runs, the value is not picked up/changed.  So I'll need to update this.

Reply:
My latest push adds in the change you suggested